pkg/apis/application/v1alpha1/generated.proto

```diff
@@ -737,6 +737,18 @@ message RepoCreds {
 
   // TLS client cert key for authenticating at the repo server
   optional string tlsClientCertKey = 6;
+
+  // Github App Private Key PEM data
+  optional string githubAppPrivateKey = 7;
+
+  // Github App ID of the app used to access the repo
+  optional int64 githubAppID = 8;
+
+  // Github App Installation ID of the installed GitHub App
+  optional int64 githubAppInstallationID = 9;
```

Comment:
This is something that is likely to change if someone removes the app from installation. I'd suggest using the repo installation api request to fetch it.

https://docs.github.com/en/rest/reference/apps#get-a-repository-installation-for-the-authenticated-app

Reply:
So the issue with this is it makes the token use and caching story somewhat complicated. In the case that an app is uninstalled and reinstalled the token would need to be used before it's known to be valid. I don't see a way to do that without adding a network call to the API every time before a token is returned to the caller. Alternatively, there would need to be a background refresh that checks installation ids but that would really complicate things.

Unlike the more traditional use of github apps where the app owner and installer are different orgs/accounts, argo's users would both be the owner and installer. Given that it would be the case and the complexities associated I don't feel adding the functionality is necessary. It's unlikely that the app would be uninstalled and re-installed in an org.

Reply:
Your installation token last a hour anyhow.
So when you request an new installation your are returned with a erroneous status code you could attempt to get the installation id.
You simply have to ask for the repository installation id as that will either be a user installation or org installation which is independent of the repo installation id.